### ISSUE TITLE
In formula arguments, ifempty cells are passed as argument, its value is 0 in desktop and null in office web

### ISSUE THREAD

**Issue by didu9898** [OfficeDev/office-js#4702]
# Provide required information needed to triage your issue

## Your Environment

* Platform [PC desktop, Mac, iOS, Office on the web]: Web & PC desktop
* Host [Excel, Word, PowerPoint, etc.]: Excel
* Office version number: ______
* Operating System: ______
* Browser (if using Office on the web): Chrome
 
## Expected behavior
The web and desktop excel behavior must be similar.

## Current behavior
Empty cells are passed as zero in desktop and null in office web.

## Steps to reproduce

```
/**
 * Check param value if empty cell is passed as param.
 * @customfunction
 * @param {any} Cell ref param
 * @returns nothing, just check console log.
 */
function checkParam(param) {
  console.log(param);
  // do nothing
  return;
}
```

This is ScriptLab code. Call checkParamfunction and pass empty cell ref as arguments and notice the param value in console log. Currently in web, the value is shown as null and in desktop the same param value is 0. Expectation is that the value should be same regardless of the platform.

## Context
I have some custom functions with cell ref as argument. The behavior is not similar in excel web and excel desktop if argument is empty cell resulting in behavior difference.

## Useful logs

Excel web:
![image](https://github.com/user-attachments/assets/bc76d9a1-4cf2-4939-b124-e4931ec52b1a)

Excel desktop:
![image](https://github.com/user-attachments/assets/e3b6938d-a103-4d58-b92a-4a8295d9ba75)

Thank you for taking the time to report an issue. Our triage team will respond to you in less than 72 hours. Normally, response time is <10 hours Monday through Friday. We do not triage on weekends.

**Comment by shanshanzheng-dev** [comment]
Hi @didu9898 Thanks for reporting this issue. But I can't repro your problem, for my side, the behavior is similar in excel web and excel desktop. They all show "null" on console log, and show "0" as result.
Desktop:
![image](https://github.com/user-attachments/assets/d543bf00-5631-4b3c-ad9b-5cd74085ac28)
Online:
![image](https://github.com/user-attachments/assets/72e5a0a0-5ab6-4f1d-ad43-44cd107a1cde)
My excel desktop version is :
![image](https://github.com/user-attachments/assets/4a5b0d4f-c194-4f13-9a49-382d1433cf3c)
for web, you can check if your custom function is registered successfully. Thanks.
![image](https://github.com/user-attachments/assets/105d9862-84de-486c-a4f1-16212b0abee3)
{
    "functions": [
        {
            "description": "Check param value if empty cell is passed as param.",
            "id": "BLANKSNIPPET1.CHECKPARAM",
            "name": "BLANKSNIPPET1.CHECKPARAM",
            "parameters": [
                {
                    "name": "param",
                    "type": "any"
                }
            ],
            "result": {}
        },
        {
            "description": "Check param value if empty cell is passed as param.",
            "id": "BLANKSNIPPET.CHECKPARAM",
            "name": "BLANKSNIPPET.CHECKPARAM",
            "parameters": [
                {
                    "name": "param",
                    "type": "any"
                }
            ],
            "result": {}
        }
    ],
    "allowCustomDataForDataTypeAny": true
}

**Comment by didu9898** [comment]
I checked and the custom function is registered successfully.

My excel version is this:
![image](https://github.com/user-attachments/assets/ab1919cc-f78b-4ef7-a87a-79a4900d9f90)

I checked with my colleagues, they also are experiencing similar behavior i.e. 0 in desktop and null in web.

**Comment by didu9898** [comment]
Can you please share screenshot with the cell formulas too?

**Comment by shanshanzheng-dev** [comment]
Desktop：=SCRIPTLAB.BLANKSNIPPET.CHECKPARAM(A1)
Web：=SCRIPTLAB.BLANKSNIPPET.CHECKPARAM(A1)
Hi @didu9898 seem our excel version is different, I'll check with my colleagues, too. Or I'm not sure if you can update a new version?

**Comment by MiaofeiWang** [comment]
Hi @didu9898 , thank you for pointing this out. This change is part of a recent improvement we implemented to address feedback from customers who wanted a way to differentiate between empty cells and cells containing the number "0" in custom functions.

This update has been fully deployed to Excel Online and is available in the Win32 version starting from build 17723.20000. This explains the discrepancy you're observing. For users on the Monthly Enterprise Channel, this feature should be available before mid-August.

Please let us know if you have any further concerns or questions. Thank you.

**Comment by didu9898** [comment]
@MiaofeiWang thank you for informing about the upcoming update.

**Comment by didu9898** [comment]
@MiaofeiWang is there a way for us to verify if a customer has received this new update? I need to find a solution that works for both customers on the old version and those on the new version.

**Comment by MiaofeiWang** [comment]
@didu9898 Unfortunately we don't have a clear way (like a new version of requirement set) to distinguish that because it was changed as a bug fix. 
But you may use some alternatives to know if it is an empty cell, for required arguments, 1) `==null` means the Excel is with the update, 2) `==0` has 2 possibilities: empty cell on old version or "0" on new/old version. To further distinguish that, you may use `CustomFunctions.Invocation` to [get the address](https://learn.microsoft.com/en-us/office/dev/add-ins/excel/custom-functions-parameter-options?tabs=javascript#invocation-parameter), and then retrieve the value by API. This approach is complex and not performant, but can be used for clients with old version.

```
    if (element === 0 && invocation.parameterAddresses[index] != undefined) {
      let range = sheet.getRange(invocation.parameterAddresses[index]).load("text");
      await context.sync();
      if (range.text[0][0] == "") {
        // '0' comes from an empty cell
      } else {
        // '0' is the real value
      }
    }
```

Hope this helps. We can discuss more if having more details regarding your scenario and requirements.

**Comment by 4tti** [comment]
@MiaofeiWang **Thanks a lot.** Yes, that's something we had as a fallback solution in case there is no better way. 

Unfortunately with this bug fix **you have introduced behavior change** which should have been communicated months ahead to have everyone prepared. **I think this should get attention and should be mentioned in the upcoming office add-ins community call...** and should be highlighted in documentation.

FYI: @jakobpn, @keyur32, @davidchesnut, @jeremy-msft

**Comment by didu9898** [comment]
I agree with @4tti, this is a significant change. I'm concerned that the fallback solution might not be acceptable for us. If a sheet has many formulas with blank cell references, recalculating the sheet would trigger multiple context.sync calls, which could greatly impact performance. This is especially problematic in large sheets, where it could cause delays and increased processing time.

**Comment by 4tti** [comment]
@didu9898  you need to do that just once in the init phase...

**Comment by MiaofeiWang** [comment]
@4tti Thank you for your suggestions. We plan to talk about this in the upcoming community call. 

@didu9898 The previous fallback solution for detecting empty input cells was the only option available before this change was implemented. We recognized its performance limitations, which is why we made this update. Unfortunately, for clients using older versions, we do not have a better workaround at this time.
